_posts/2024-04-10-benchmarks-over-time.md

```diff
@@ -0,0 +1,547 @@
+
+<div id="overall_results_by_version" style="width:100%;height:400px;"></div>
+<script>
+    fetch('{{ site.baseurl }}/data/perf_over_time_overall_results_by_version.json')
+        .then(res => res.json())
+        .then(parsed_json => {
+            let overall_results_by_version = document.getElementById('overall_results_by_version');
+            parsed_json.layout = {...parsed_json.layout, "title": "Benchmark results by version"};
+            Plotly.plot( overall_results_by_version, parsed_json.data, parsed_json.layout );
+            });
+</script>
+
+If you remember the version that you last tested, you can compare how much faster things are now with 1.0!
+
+<div id="overall_results_by_version_relative" style="width:100%;height:400px;"></div>
+<script>
+    fetch('{{ site.baseurl }}/data/perf_over_time_overall_results_by_version_relative.json')
+        .then(res => res.json())
+        .then(parsed_json => {
+            let overall_results_by_version_relative = document.getElementById('overall_results_by_version_relative');
+            parsed_json.layout = {...parsed_json.layout, "title": "Relative benchmark results by version"};
+            Plotly.plot( overall_results_by_version_relative, parsed_json.data, parsed_json.layout );
+            });
+</script>
+
+## Results by Workload
+
+### CSV Reader
+
+<div id="perf_over_time_csv_reader_area" style="width:100%;height:400px;"></div>
+<script>
+    fetch('{{ site.baseurl }}/data/perf_over_time_csv_reader_area.json')
+        .then(res => res.json())
+        .then(parsed_json => {
+            let perf_over_time_csv_reader_area = document.getElementById('perf_over_time_csv_reader_area');
+            Plotly.plot( perf_over_time_csv_reader_area, parsed_json.data, parsed_json.layout );
+            });
+</script>
+
+DuckDB has invested substantially in building a [fast and robust CSV parser](https://duckdb.org/2023/10/27/csv-sniffer.html).
+This is often the first task in a data analysis workload, and it tends to be under valued and under-benchmarked. 
+DuckDB has **improved csv reader performance by nearly 3x**, while adding the ability to handle many more csv file formats automatically.
+
+### Group By
+
+<div id="perf_over_time_group_by_area" style="width:100%;height:400px;"></div>
+<script>
+    fetch('{{ site.baseurl }}/data/perf_over_time_group_by_area.json')
+        .then(res => res.json())
+        .then(parsed_json => {
+            let perf_over_time_group_by_area = document.getElementById('perf_over_time_group_by_area');
+            Plotly.plot( perf_over_time_group_by_area, parsed_json.data, parsed_json.layout );
+            });
+</script>
+
+Group by or aggregation operations are a critical OLAP workload, and have therefore received substantial focus in DuckDB, **improving over 12x in the last 3 years**.
+
+In version 0.3.1, multithreaded aggregation was enabled by default, providing a significant speedup.
```

Comment:
I can do that, sure! I had to make a mapping of it for the script anyway. The first draft had used a by-version chart instead of by time, but I should have updated the language!

Reply:
Made these changes, thank you!